### PR TITLE
refactor events to new package

### LINKED ISSUE #416: Refactor events to new package
The following event classes are within `ReadingGoalForm`:

```java
    public <T extends ComponentEvent<?>> Registration addListener(Class<T> eventType,
                                                                  ComponentEventListener<T> listener) {
        return getEventBus().addListener(eventType, listener);
    }

    /**
     * Vaadin's event bus system. A registered listener can be notified when a save or delete event
     * is fired
     */
    public abstract static class GoalFormEvent extends ComponentEvent<ReadingGoalForm> {
        private final ReadingGoal readingGoal;

        protected GoalFormEvent(ReadingGoalForm source, ReadingGoal readingGoal) {
            super(source, false);
            this.readingGoal = readingGoal;
        }

        public ReadingGoal getReadingGoal() {
            return readingGoal;
        }
    }

    public static class SaveEvent extends GoalFormEvent {
        SaveEvent(ReadingGoalForm source, ReadingGoal readingGoal) {
            super(source, readingGoal);
        }
    }

    // TODO: implement deleting reading goal
    public static class DeleteEvent extends GoalFormEvent {
        DeleteEvent(ReadingGoalForm source, ReadingGoal readingGoal) {
            super(source, readingGoal);
        }
    }
```

Refactor this into a new `events` package inside the `ui/goal` package. Each class should be in its own file.

### PR DESCRIPTION
**Pull request by mareksmelko777**
## Summary of change

I refactored it inside goal and created new file classes for each class.

## Related issue

Closes #416 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ)

### CODE REVIEW

**Comment by mareksmelko777** [comment]
I did it, could you check it please? Thanks

**Comment by knjk04** [comment]
@mareksmelko777 Thank you very much for working on this! We'll review this shortly

**Comment by knjk04** [comment]
@mareksmelko777 Apologies for the delay in reviewing this. Thanks for adding the new classes. However, I noticed that there's no implementation and that the event classes are still in the `ReadingGoalForm`. Could you please move their implementation to the new classes you created?

I'll mark this PR as a draft. When you're ready for a review, feel free to request one by clicking the 'ready for review' button below.

Lastly, as there's not a lot of time left until the end of the month, I'd recommend addressing the feedback quickly (if possible) in order to increase the probability that we'll be able to get this merged in before the 31st of October

**Comment by mareksmelko777** [comment]
@knjk04 Thanks for your respond, I will try my best to solve it. Thank you very much for the patience.

**Comment by knjk04** [comment]
@mareksmelko777 If you need anything, we're here to help on Slack

**Comment by mareksmelko777** [comment]
@knjk04 It's alright for now, If I will need everything, I write you on slack :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ## Additional context
 
-[If a screenshot would be useful (e.g. for UI changes), please provide one.]
+[If a screenshot would be useful (e.g. for UI changes), please provide one. If not, please remove this section]
```
Comment (knjk04) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: Good idea

File: src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java [hunk 2]
```diff
@@ -107,7 +107,7 @@ private void updateSetGoalText() {
         setGoalButton.setText(UPDATE_GOAL);
     }
 
-    private void saveGoal(ReadingGoalForm.SaveEvent event) {
+    private void saveGoal(ReadingGoalForm.SaveGoalEvent event) {
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java:110]: `ReadingGoalForm` does not have an inner class called `SaveGoalEvent`. Could you please remove `ReadingGoalForm.` here?

File: src/main/java/com/karankumar/bookproject/ui/goal/DeleteGoalEvent.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.karankumar.bookproject.ui.goal;
+
+import com.karankumar.bookproject.backend.entity.ReadingGoal;
+public class DeleteGoalEvent {
+    // TODO: implement deleting reading goal
+    public static class DeleteEvent extends ComponentGoalEvent.GoalFormEvent {
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/DeleteGoalEvent.java:6]: Could you please remove the inner class and make the current outer class extend `ComponentGoalEvent.GoalFormEvent`. Note that a constructor will be needed

File: src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.karankumar.bookproject.ui.goal;
+
+import com.karankumar.bookproject.backend.entity.ReadingGoal;
+
+public class SaveGoalEvent {
+    public static class SaveEvent extends ComponentGoalEvent.GoalFormEvent {
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java:6]: As with `DeleteEvent`, could you please remove the inner class here and make the outer class extend `ComponentGoalEvent.GoalFormEvent`?

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for working on this. As requested, I've left some feedback that you should help you get this to compile

**Comment by mareksmelko777** [comment]
@knjk04 I finally did it, big thanks to you for helping me with this issue. I builded it and made some more little changes and it's witout errors now. Thanks for the review.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -2,9 +2,12 @@
 
 [Please briefly describe what you have changed.]
 
+
+
 ## Additional context
 
-[If a screenshot would be useful (e.g. for UI changes), please provide one. If not, please remove this section]
```
Comment (knjk04) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: Could you please bring this line back?

File: src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    private final ReadingGoal readingGoal;
+
+    protected GoalFormEvent(ReadingGoalForm source, ReadingGoal readingGoal) {
+        super(source, false);
+        this.readingGoal = readingGoal;
+    }
+
+    public ReadingGoal getReadingGoal() {
+        return readingGoal;
+    }
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java:17]: This section should be indented four space

File: src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+}
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java:18]: This curly brace should be indented four spaces

File: src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package com.karankumar.bookproject.ui.goal;
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java:1]: In line with our style guide, could you please attach our [copyright notice](https://github.com/Project-Books/book-project/blob/master/docs/COPYRIGHT) to the start of this file?

File: src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package com.karankumar.bookproject.ui.goal;
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java:1]: As before, could you please add our copyright notice to the start of this file?

File: src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.karankumar.bookproject.backend.entity.ReadingGoal;
+
+public class SaveGoalEvent extends ComponentGoalEvent.GoalFormEvent{
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java:5]: There should be a space before the curly brace

File: src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+
+        SaveGoalEvent(ReadingGoalForm source, ReadingGoal readingGoal) {
+            super(source, readingGoal);
+        }
+    }
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java:10]: This last curly brace is incorrectly indented. It should be four spaces to the left

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for fixing the compile-time errors. This now works as intended. 

I've left some code style-related feedback for you to address. 

Also, I've removed the `DeleteGoalEvent` class as we are not currently using it.

File: src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java (hunk 6 above)
Comment (mareksmelko777) [reply src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java:17]: @knjk04 every line shoud have 4 spaces?

Comment (knjk04) [reply src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java:17]: @mareksmelko777 Every line should be indented four spaces from the previous indentation level. 

Here, you've indented the `GoalFormEvent` class line. That's fine. However, the body of the inner class should also be indented.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (mareksmelko777) [reply .github/PULL_REQUEST_TEMPLATE.md]: @knjk04 Hello, I did everything except this, pushed and tests passed, last step I add this line back and tests not passed. What should I do now?

**Comment by mareksmelko777** [comment]
@knjk04 I think I fixed it. Could you check and review it please? If something is still wrong, feel free and write me, I will try to fix it. Thank you very much.

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for addressing the style changes.

I noticed one more thing after looking at the issue. I noticed that you did not create an `events` package and place the `ComponentGoalEvent` and `SaveGoalEvent` inside. Could you please do this? Once done, this will be ready to be merged

**Comment by mareksmelko777** [comment]
@knjk04 I'm sorry I forgot to create the package, I fixed it. I hope it's alright now :) .

**Comment by mareksmelko777** [comment]
@knjk04 I created it in goal package. I hope it's fine. Thank you.

File: src/main/java/com/karankumar/bookproject/ui/goal/DeleteGoalEvent.java (hunk 3 above)
Comment (knjk04) [reply src/main/java/com/karankumar/bookproject/ui/goal/DeleteGoalEvent.java]: I've deleted this

File: src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java (hunk 4 above)
Comment (knjk04) [reply src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java]: I think you forgot to do this, so I'll add this in

**Review by knjk04 — APPROVED** [review]
Thanks for working on this!

**Pull request merged**